core-addon/IonCore.js

```diff
@@ -89,6 +89,11 @@ module.exports = class IonCore {
         // is expecting it.
         return this._enrollStudy(message.data.studyID).then(r => true);
       } break;
+      case "study-unenrollment": {
```

Comment:
@Dexterp37 I added this case here so we can handle the unenrollment, though of course we cannot do this from the core add-on itself. Is this ok?